pkg/agent/cridockerd/cridockerd.go

```diff
@@ -32,6 +35,10 @@ func Run(ctx context.Context, cfg *config.Node) error {
 		logrus.Fatalf("cri-dockerd exited: %v", command.ExecuteContext(ctx))
 	}()
 
+	if err := cri.WaitForService(ctx, cfg.CRIDockerd.Address, "cri-dockerd"); err != nil {
```

Comment:
This can be simplified a bit further.
```suggestion
	return cri.WaitForService(ctx, cfg.CRIDockerd.Address, "cri-dockerd")
```